subsys/net/lib/tls_credentials/tls_credentials_shell.c

```diff
@@ -0,0 +1,813 @@
+		if (type_strings[i].type == type) {
+			return type_strings[i].name;
+		}
+	}
+
+	/* No matches found, it's invalid. */
+	return "INVALID";
+}
+
+/* Get the compact name of a TLS credential type*/
+static char* cred_type_name_compact(enum tls_credential_type type)
+{
+	/* Scan over predefined type strings, and return the name
+	 * of the last one of matching type.
+	 */
+	for (int i = NUM_TYPE_STRINGS - 1; i >= 0; i--) {
+		if (type_strings[i].type == type) {
+			return type_strings[i].name;
+		}
+	}
+
+	/* No matches found, it's invalid. */
+	return "INV";
+}
+
+/* Shell interface routines */
+
+/* Attempt to parse a command line argument into a sectag.
+ * TLS_SEC_TAG_NONE is returned if ANY_KEYWORD is provided.
+ */
+static int shell_parse_cred_sectag(const struct shell *sh, char *arg, sec_tag_t *out,
+				   bool allow_any)
+{
+
+	unsigned long sectag_value;
+	int err = 0;
+
+	/* Check for "ANY" special keyword if desired. */
+	if (allow_any && strcasecmp(arg, ANY_KEYWORD) == 0) {
+		*out = TLS_SEC_TAG_NONE;
+		return 0;
+	}
+
+	/* Otherwise, verify that the sectag is purely numeric */
+	if (!check_numeric(arg)) {
+		err = -EINVAL;
+		goto error;
+	}
+
+	/* Use strtoul because it has nicer validation features than atoi */
+	sectag_value = shell_strtoul(arg, 10, &err);
+
+	if (!err) {
+		*out = (sec_tag_t)sectag_value;
+		return 0;
+	}
+
+error:
+	shell_fprintf(
+		sh, SHELL_ERROR,
+		"%s is not a valid sectag.\n", arg
+	);
+
+	return err;
+}
+
+/* Attempt to parse a command line argument into a credential type.
+ * TLS_CREDENTIAL_NONE is returned if ANY_KEYWORD is provided.
+ */
+static int shell_parse_cred_type(const struct shell *sh, char *arg, enum tls_credential_type *out,
+				 bool allow_any)
+{
+	/* Check for "ANY" special keyword if desired. */
+	if (allow_any && strcasecmp(arg, ANY_KEYWORD) == 0) {
+		*out = TLS_CREDENTIAL_NONE;
+		return 0;
+	}
+
+	/* Otherwise, scan over predefined type strings, and return the corresponding
+	 * credential type if one is found
+	 */
+	for (int i = 0; i < NUM_TYPE_STRINGS; i++) {
+		if (strcasecmp(arg, type_strings[i].name) == 0) {
+			*out = type_strings[i].type;
+			return 0;
+		}
+	}
+
+	/* No matches found, it's invalid. */
+	shell_fprintf(
+		sh, SHELL_ERROR,
+		"%s is not a valid credential type.\n", arg
+	);
+
+	return -EINVAL;
+}
+
+/* Parse a backend specifier argument
+ * Right now, only a single backend is supported, so this is serving simply as a reserved argument.
+ * As such, the only valid input is "default"
+ */
+static int shell_parse_cred_backend(const struct shell *sh, char *arg)
+{
+	if (strcasecmp(arg, "default") == 0) {
+		return 0;
+	}
+
+	shell_fprintf(
+		sh, SHELL_ERROR,
+		"%s is not a valid backend.\n", arg
+	);
+
+	return -EINVAL;
+}
+
+/* Parse an input type specifier */
+static int shell_parse_cred_storage_format(const struct shell *sh, char *arg,
+					   enum cred_storage_fmt *out, bool *terminated)
+{
+	if (strcasecmp(arg, "bin")	== 0) {
```

Comment:
I suggest not putting extra spaces here and below, it just looks weird.

Reply:
You're right, not sure what I was thinking there